Added typing

BiffoBear opened the pull request:
Added typing to all modules. Made two small edits to improve typing compatibility.

In `adafruit_wiznet5k.adafruit_wiznet5k_ntp.__init__()` changed `utc: int` to `utc: float` as some time zones are on the half hour. Added `int()` to  `get_time()` to allow for a float argument being passed.

In `adafruit_wiznet5k.adafruit_wiznet5k_dns.__init__()` changed `self._host = 0` to `self._host = b""` to remove a warning about it not having a `to_int()` method. `self._host` is set to a `bytes` object in `gethostbyname()` and is read as a bytes object in `_build_dns_question()` but nowhere else.

I am confident that these changes don't break the code.

There are three other type conflicts that are more complex and have not been addressed.

tekktrik commented:
This is a big one, so give us some time to look over this, but thanks for doing it!  Heads up that it may be easier to do this review in a few rounds, tackling different things each time, so its more manageable.

BiffoBear commented:
I expect it to take a few iterations, I'm expecting to have to make some changes. Please take your time.

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k_socket.py:
```diff
@@ -88,17 +139,28 @@ def is_ipv4(host):
+        type: int = SOCK_STREAM,
+        proto: Any = 0,
+        fileno: Optional[int] = None,
+        socknum: Optional[int] = None,
```
`fileno` and `socknum` are typed `int` here in the arguments. But down a few lines in the docstring they are `Any`. 

Since they are unused it probably doesn't matter too much, but I think it'd be best to make them consistent. Can you change one or the other so that they match?

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k_socket.py:
```diff
@@ -255,30 +360,40 @@ def connect(self, address, conntype=None):
+        # pylint: disable=too-many-branches
+        self,
+        bufsize: int = 0,
+        flags: Any = 0,
```
I think it'd be best to make the type declaration here of `flags` match the one in the doc strings. I don't know that it matters much if we use `int` or `Any` since it's ignored. Can you change one or the other so they are consistent.

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k_socket.py:
```diff
@@ -335,10 +450,17 @@ def recv(self, bufsize=0, flags=0):  # pylint: disable=too-many-branches
-        """Reads some bytes from the connected remote address and then return recv().
-        :param int bufsize: Maximum number of bytes to receive.
+    def embed_recv(
+        self, bufsize: int = 0, flags: Any = 0
```
another spot where I think we should match the `flags` type declaration with docstrings.

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k_socket.py:
```diff
@@ -385,13 +514,18 @@ def recv_into(self, buf, nbytes=0, flags=0):
-    def recvfrom_into(self, buf, nbytes=0, flags=0):
-        """Reads some bytes from the connected remote address info the provided buffer.
+    def recvfrom_into(
+        self, buf: bytearray, nbytes: int = 0, flags: Any = 0
```
another place I think we should match `flags` type declaration with the docstrings.

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k_socket.py:
```diff
@@ -401,9 +535,14 @@ def recvfrom_into(self, buf, nbytes=0, flags=0):
+        Read a line from the socket.
+
+        Attempt to return as many bytes as we can up to but not including a carriage return and
+        linefeed charater pair.
```
small typo I think missing the 'c'. Should be `character`

FoamyGuy commented on adafruit_wiznet5k/adafruit_wiznet5k.py:
```diff
@@ -947,7 +1143,7 @@ def _write_socket(self, sock, address, data):
         return None
 
-    def _read_socket(self, sock, address):
+    def _read_socket(self, sock: int, address: int) -> Optional[bytearray]:  # *1
```
is the `# *1` comment something leftover or does it have some meaning for the types?

BiffoBear commented on adafruit_wiznet5k/adafruit_wiznet5k_ntp.py:
```diff
@@ -17,24 +17,38 @@
+        self,
+        iface: WIZNET5K,
+        ntp_address: str,
+        utc: float,
```
float because not all timezones are whole hours.

BiffoBear commented:
@FoamyGuy Thanks for your kind and thorough review. Requested changes made. I've also removed all the 'Any' types and replaced them with the types expected by the CPython equivalents.

FoamyGuy reviewed (approved):
Looks good to me. Tested successfully with simpletest example on Feather S2 TFT 8 beta4.

Thank you for taking this on @BiffoBear!
